PyTDF/TDataFrame.py

```diff
@@ -17,15 +17,16 @@ def __init__(self, treename, filelist):
         super(TDataFrame, self).__init__(None, None)
         self.filelist = filelist
         self.treename = treename
-        self.chain = ROOT.TChain(treename)
-        self._tdf = self.get_tdf_temp()
 
-    def get_tdf_temp(self):
-        # TODO (shravan97) : shift this to mapper generator
-        # Temporary function to get 
-        # TDF object until mapper is done
+    def create_tdf(self):
```

Comment:
The creation of the C++ dataframe should not be the job of pyTDF but the backend's.
In distributed executions, for example, the backend will instruct each worker to create its own C++ dataframe.

Reply:
This function call is triggered only inside the backend. @bluehood , you'd mentioned earlier that the TDF head node has to be passed around in order for it to work. So, I'm calling this function inside the backend only

Reply:
The requirement is that the TDF head node must be in scope when you run the event loop.
You can (should) create the head node in the backend and keep it alive until you ran the event loop.
Note that in distributed executions worker nodes probably will not have a handle to the pyTDF head node so they wouldn't be able to create a C++ TDF head node from it.

Reply:
Also, once the bleeding edge version of ROOT is released, I'll remove this and make it inside backend completely. In that case, I don't have to pass the TDF head node around

Reply:
the requirement that the TDF head node must be in scope when you run the event loop will be there in 6.14 (the upcoming ROOT release) so pyTDF will have to live with it I'm afraid (I didn't manage to get this fixed in time for 6.14).

Reply:
@bluehood point is indeed what we want, the real TDF only lives inside the backend. The local backend will create it locally, distributed backends will create it on the worker nodes for a given range.